### ISSUE TITLE
Location header

### ISSUE THREAD

**Issue by gottfrois** [jpscaletti/tus#2]
Thanks for the great work on the package! I've noticed the `Location` header only includes the file `uid` whereas according to the protocol, it should include the full URL:

```
HTTP/1.1 201 Created
Location: https://tus.example.org/files/24e533e02ec3bc40c387f1a0e460e216
Tus-Resumable: 1.0.0
```

**Comment by ringods** [comment]
@gottfrois I got it working when patching `post.ex` to have line 15 as:

```
      |> put_resp_header("location", conn.request_path <> file.uid)
```

It needs a bit more fine-tuning to check whether `request_path` already has a trailing `/`. 

I can add the `host` info too, but I only miss out on the correct protocol (http/https) in the `Plug.Conn` structure. Any idea how to find out correctly, both with direct connections as well as behind a proxy?

**Comment by gottfrois** [comment]
Yes, I think we need something more sophisticated especially to handle the `x-forwarded` headers when the service is behind a proxy. I don't have any inputs at the moment unfortunately but I think you can find great inspiration from the official Go server implementation [https://github.com/tus/tusd](https://github.com/tus/tusd)

**Comment by ringods** [comment]
OK, I think I can continue based on these Go snippets:

https://github.com/tus/tusd/blob/master/unrouted_handler.go#L796-L809

https://github.com/tus/tusd/blob/master/unrouted_handler.go#L855-L891

**Comment by gottfrois** [comment]
Yes! Exactly what I had in mind regarding the `x-forwarded` headers
